### PR TITLE
Add group features to iot

### PR DESCRIPTION
**Pull request by toshitanian**
Added group related features to aws-iot. @terrycain @JackDanger please review if you time ;)

### CODE REVIEW

File: moto/iot/models.py [hunk 1]
```diff
@@ -44,6 +45,7 @@ def __init__(self, thing_type_name, thing_type_properties, region_name):
         self.region_name = region_name
         self.thing_type_name = thing_type_name
         self.thing_type_properties = thing_type_properties
+        self.thing_type_id = str(uuid.uuid4())  # I don't know the rule of id
```
Comment (JackDanger) [inline moto/iot/models.py]: The docs all just say 'string' so what you've done seems very reasonable

File: moto/iot/responses.py [hunk 2]
```diff
@@ -63,7 +63,6 @@ def describe_thing(self):
         thing = self.iot_backend.describe_thing(
             thing_name=thing_name,
         )
-        print(thing.to_dict(include_default_client_id=True))
```
Comment (JackDanger) [inline moto/iot/responses.py]: Nice catch!

File: moto/iot/responses.py [hunk 3]
```diff
@@ -256,3 +255,123 @@ def list_thing_principals(self):
...
+        thing_group_arn = self._get_param("thingGroupArn")
+        thing_name = self._get_param("thingName")
+        thing_arn = self._get_param("thingArn")
+        self.iot_backend.remove_thing_from_thing_group(
+            thing_group_name=thing_group_name,
+            thing_group_arn=thing_group_arn,
+            thing_name=thing_name,
+            thing_arn=thing_arn,
+        )
+        # TODO: adjust response
```
Comment (JackDanger) [inline moto/iot/responses.py]: Are these TODOs something that you can do now or are we blocked on something?

Comment (toshitanian) [reply moto/iot/responses.py]: @JackDanger thanks for your review. I removed meaningless comments and added TODO comments for future implementations. This should be readly to merge ;)

**Pull request merged**